components/scream/src/share/grid/remap/horizontal_remap_utility.hpp

```diff
@@ -0,0 +1,163 @@
+#ifndef EAMXX_HORIZONTAL_REMAP_UTILITY_HPP
+#define EAMXX_HORIZONTAL_REMAP_UTILITY_HPP
+
+#include "share/grid/abstract_grid.hpp"
+#include "share/io/scorpio_input.hpp"
+#include "share/scream_types.hpp"
+#include "ekat/kokkos/ekat_subview_utils.hpp"
+
+#include <numeric>
+
+namespace scream {
+
+/*===============================================================================================*/
+/*                                        GSSegment
+ * Lightwieght structure to represent a single remapping of source data to a single column.
```

Comment:
Lightwieght -> Lightweight. Also, worth saying "single target column" instead of "single column"?